### ISSUE TITLE
Make the new adapter interface consistent

### ISSUE THREAD

**Issue by ZackButcher** [istio/mixer#1132]
Today we have these two methods:

```go
// map name->Type
ConfigureFooHandler(foos map[string]*foo.Type) error
// foo.Instance has a "Name" field
HandleFoo(context.Context, []*foo.Instance) error
```

We should be consistent: either `foo.Type` needs a `Name` field and we change the signature of `ConfigureFooHandler` to `ConfigureFooHandler(foos []*foo.Type) error`, or we need to pull the `Name` field out of `foo.Instance` and change `HandleFoo` to `HandleFoo(context.Context, map[string]*foo.Instance)`. As-is the asymmetry feels inconsistent.

cc @guptasu @geeknoid

**Comment by geeknoid** [comment]
The second form would need to be:

HandleFoo(context.Context, map[string][]*foo.Instance)

since there can be multiple instances for a given type.

**Comment by geeknoid** [comment]
I disagree that there is an issue here. The data being provided is in the most directly meaningful form for the adapter author.

I do think we should add a Name field to the Type objects such that Type objects can be self-contained.

**Comment by geeknoid** [comment]
Can this be closed?

**Comment by guptasu** [comment]
We need to add Name to Type too. I need to fix that.
